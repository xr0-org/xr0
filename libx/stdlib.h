#ifndef STDLIB_H
#define STDLIB_H

axiom void *
malloc(int size) [ .alloc result; ];

axiom void
free(void *ptr) [ .dealloc ptr; ];

#endif
