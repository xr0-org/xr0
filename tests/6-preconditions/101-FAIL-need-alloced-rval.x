#include <stdlib.h>

int
func(int *x) ~ [ 
	int i;
	pre: {
		.alloc x;
		*x = $;
	}
	result = *x;
]{
	return *x;	
}

int
main()
{
	int p;
	int q;
	p = malloc(1);
	q = func(p);	/* ERROR: spec required rvalue */
}
