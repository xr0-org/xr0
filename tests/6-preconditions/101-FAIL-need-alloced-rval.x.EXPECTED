6-preconditions/101-FAIL-need-alloced-rval.x:21:13: cannot exec statement: parameter 'x' must be rvalue
